src/include/miopen/generic_search.hpp

```diff
@@ -381,9 +381,26 @@ auto GenericSearch(const Solver s,
     std::random_device rd{};
     auto rng = std::default_random_engine{rd()};
     std::shuffle(all_configs.begin(), all_configs.end(), rng);
-    const std::size_t n_runs_total = std::min(all_configs.size(), GetTuningIterationsMax());
+    std::size_t n_runs_total = std::min(all_configs.size(), GetTuningIterationsMax());
     all_configs.resize(n_runs_total);
 
+    if(all_configs.empty())
+    {
+        const auto default_config = s.GetDefaultPerformanceConfig(context, problem);
+
+        if(default_config.IsValid(context, problem))
+        {
+            all_configs.emplace_back(default_config);
+            n_runs_total += 1;
+        }
+        else
```

Comment:
@DrizztDoUrden @shurale-nkn @JehandadKhan This change is not necessary to resolve #2270 and seems excessive to me. Why:

If the Solver produces none performanceConfigs for tuning (container is empty), then something is definitely wrong and needs to be fixed (e.g. applicability of the solver needs to be corrected or so). If we manually add one default config, then we just **_camouflage the problem_** and waste resources. Because as a result of tuning, the library will write the default performance config into the perf-db. 

If we simply fail tuning and keep perf-db intact, the solver will use the default config later anyway. This way the library will be protected against the "empty container" problem while keeping the database free of junk.

I am going to fix this soon.

### Some details

The container is expected to hold all the potentially usable performanceConfigs, including the default one. Therefore the default performance config is expected to reside in the container (either primary of spare). If this is not so, then the Solver must be fixed. In other words, the current contract does not allow situations when container is empty while the default performance config is valid.

If we would like to change the contract and allow Solvers to produce empty containers (but and allow the default performance config to be valid (which I do not like), then we should seriously consider splitting the Solver to two: оne is tunable, another one is not. That is necessary to optimize out unnecessary database accesses.

Reply:
[won't fix] see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2296#issuecomment-1680731004